File: Data Structures & Algorithms/number-of-one-bits/submission-9.java
class Solution {
    public int hammingWeight(int n) {
        int res = 0;

        for (int i = 0; i < 32; i++) {
            int lastBit = n & 1;

            if (lastBit == 1) {
                res += 1;
            }

            n >>= 1;
        }

        return res;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-9.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        #maxheap version 
        res = []
        counterdict = {} 
        for num in nums:
            counterdict[num] = counterdict.get(num, 0) + 1 

        maxheap = [] 
        for key,value in counterdict.items():
            maxheap.append([-value, key])

        #ok so maxheap now contains list of lists with first element being negative value 

        i = 0 
        while i < k: 
            res.append(maxheap.pop()[1])
            i += 1 

        return res 